localstack/testing/testselection/matching.py

```diff
@@ -0,0 +1,156 @@
+import fnmatch
+import re
+from typing import Callable, Iterable
+
+from localstack.aws.scaffold import is_keyword
+
+# TODO: extract API Dependencies and composites to constants or similar
+from localstack.utils.bootstrap import API_DEPENDENCIES
+
+SENTINEL_NO_TEST = "SENTINEL_NO_TEST"  # a line item which signals that we don't default to everything, we just don't want to actually want to run a test => useful to differentiate between empty / nothing
+SENTINEL_ALL_TESTS = "SENTINEL_ALL_TESTS"  # a line item which signals that we don't default to everything, we just don't want to actually want to run a test => useful to differentiate between empty / nothing
+
+
+def _map_to_module_name(service_name: str) -> str:
+    """sanitize a service name like we're doing when scaffolding, e.g. lambda => lambda_"""
+    service_name = service_name.replace("-", "_")
+    # handle service names which are reserved keywords in python (f.e. lambda)
+    if is_keyword(service_name):
+        service_name += "_"
+    return service_name
+
+
+def _map_to_service_name(module_name: str) -> str:
+    """map a sanitized module name to a service name, e.g. lambda_ => lambda"""
+    if module_name.endswith("_"):
+        return module_name[:-1]
+    return module_name.replace("_", "-")
+
+
+def resolve_dependencies(module_name: str) -> set[str]:
```

Comment:
This logic actually needs to be reversed. We need to additionally test the services that depend *on* the changed service, so we need to figure out which other services depend on this one.